resources/js/chat/conversation-view.tsx

```diff
@@ -259,16 +259,33 @@ export default class ConversationView extends React.Component<Props> {
   renderUsers() {
     if (this.currentChannel?.type !== 'ANNOUNCE') return null;
 
+    const users = this.currentChannel.users;
+
+    if (users != null && users.length === 0) {
+      return null;
+    }
+
     return (
-      <div className={classWithModifiers('chat-conversation__users', { loading: this.currentChannel.announcementUsers == null })}>
-        {this.currentChannel.announcementUsers == null ? (
-          <>
-            <Spinner modifiers='self-center' /><span>{trans('chat.loading_users')}</span>
-          </>
-        ) : (
-          this.currentChannel.announcementUsers.map((user) => (
-            <UserCardBrick key={user.id} user={user.toJson()} />
-          ))
+      <div className='chat-conversation__users-container'>
+        <div className={classWithModifiers('chat-conversation__users', { loading: users == null })}>
+          {users == null ? (
+            <>
+              <Spinner modifiers='self-center' /><span>{trans('chat.loading_users')}</span>
+            </>
```

Comment:
I think we can just skip trying to load users if they don't have permission to view it anyway. 
Or maybe return `users: []` with the channel if they have no permission to load users and `null` otherwise 🤔  (or both)

Reply:
I'll look into that... eventually...

probably not changing existing json fields though.